src/main/kotlin/com/manimdsl/frontend/SymbolTableVisitor.kt

```diff
@@ -1,16 +1,20 @@
 package com.manimdsl.frontend
 
-open class IdentifierData(val type: Type)
+open class IdentifierData(open val type: Type)
 
 object ErrorIdentifierData : IdentifierData(ErrorType)
 
+data class FunctionData(val parameters: List<ParameterNode>, override val type: Type) : IdentifierData(type)
```

Comment:
Maybe create a sealed class/interface called something like `Data`/`SymbolTableData` and have `IdentifierData` and `FunctionData` both extend that? It can have a required `type` attribute